subprojects/model-core/src/main/java/org/gradle/internal/reflect/validation/TypeValidationProblemRenderer.java

```diff
@@ -122,7 +120,7 @@ public static String introductionFor(Map<String, String> additionalMetadata) {
         String rootType = ofNullable(additionalMetadata.get(TYPE_NAME))
             .filter(TypeValidationProblemRenderer::shouldRenderType)
             .orElse(null);
-        ProblemsPluginId pluginId = ofNullable(additionalMetadata.get(PLUGIN_ID)).map(DefaultPluginId::new).orElse(null);
+        DefaultPluginId pluginId = ofNullable(additionalMetadata.get(PLUGIN_ID)).map(DefaultPluginId::new).orElse(null);
```

Comment:
`ProblemsPluginId` was part of the API surface but was only used in the type validation renderer. I removed it from the API and inlined the classes here.